Makefile

```diff
@@ -90,6 +92,22 @@ test-silent:
 benchmark-test:
 	. ./scripts/shared_env && go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
 
+test-artifacts:
+	mkdir -p ./out/test-artifacts
+	go test -race -tags integration -o ./out/test-artifacts/unix-engine-tests -c ./agent/engine
+	go test -race -tags integration -o ./out/test-artifacts/unix-stats-tests -c ./agent/stats
+	go test -race -tags integration -o ./out/test-artifacts/unix-app-tests -c ./agent/app
+	go test -tags functional -o ./out/test-artifacts/unix-simple-tests -c ./agent/functional_tests/tests/generated/simpletests_unix/
+	go test -tags functional -o ./out/test-artifacts/unix-handwritten-tests -c ./agent/functional_tests/tests/
+
+	# Race detector on windows requires CGO
+	GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go test -race -tags integration -o ./out/test-artifacts/windows-engine-tests.exe -c ./agent/engine
+	GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go test -race -tags integration -o ./out/test-artifacts/windows-stats-tests.exe -c ./agent/stats
+	GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go test -race -tags integration -o ./out/test-artifacts/windows-app-tests.exe -c ./agent/app
+	GOOS=windows go test -tags functional -o ./out/test-artifacts/windows-simple-tests.exe -c ./agent/functional_tests/tests/generated/simpletests_windows/
+	GOOS=windows go test -tags functional -o ./out/test-artifacts/windows-handwritten-tests.exe -c ./agent/functional_tests/tests/
+	GOOS=windows go build -o out/test-artifacts/agent.exe ./agent
```

Comment:
I'd prefer to use the same [build script](https://github.com/aws/amazon-ecs-agent/blob/v1.16.1/scripts/build) for the actual agent that we use elsewhere.

Reply:
Doing that now. I hadn't addressed your set of comments before pushing my updates.